lib/pharos/addon.rb

```diff
@@ -114,13 +114,13 @@ def validate(config)
 
     attr_reader :config, :cpu_arch, :cluster_config, :master
 
-    # @param config [Hash]
+    # @param config [Hash,Dry::Validation::Result]
     # @param enabled [Boolean]
     # @param master [Pharos::Configuration::Host,NilClass]
     # @param cpu_arch [String, NilClass]
     # @param cluster_config [Pharos::Config, NilClass]
     def initialize(config = nil, enabled: true, master:, cpu_arch:, cluster_config:)
-      @config = self.class.config? ? self.class.config.new(config) : RecursiveOpenStruct.new(config)
+      @config = self.class.config? ? self.class.config.new(config) : RecursiveOpenStruct.new(config.to_hash)
```

Comment:
Potential for `NoMethodError (undefined method `to_hash' for nil:NilClass)` because config defaults to nil.

Can be avoided by `RecursiveOpenStruct.new(Hash(config))` (returns an empty hash or tries .to_hash)